hercules/plant_components/wind_farm_scada_power.py

```diff
@@ -0,0 +1,346 @@
+        self.n_turbines = len(self.power_columns)
+
+        self.logger.info(f"Inferred number of turbines: {self.n_turbines}")
+
+        # Collect the turbine powers
+        self.scada_powers = df_scada[self.power_columns].to_numpy(dtype=hercules_float_type)
+
+        # Now get the wind speed and directions
+
+        # Convert the wind directions and wind speeds and ti to numpy matrices
+        if "ws_mean" in df_scada.columns and "ws_000" not in df_scada.columns:
+            self.ws_mat = np.tile(
+                df_scada["ws_mean"].values.astype(hercules_float_type)[:, np.newaxis],
+                (1, self.n_turbines),
+            )
+        else:
+            self.ws_mat = df_scada[
+                [f"ws_{t_idx:03d}" for t_idx in range(self.n_turbines)]
+            ].to_numpy(dtype=hercules_float_type)
+
+        # Compute the turbine averaged wind speeds (axis = 1) using mean
+        self.ws_mat_mean = np.mean(self.ws_mat, axis=1, dtype=hercules_float_type)
+
+        self.initial_wind_speeds = self.ws_mat[0, :]
+        self.wind_speed_mean_background = self.ws_mat_mean[0]
+
+        # Get the initial background wind speeds
+        self.wind_speeds_background = self.ws_mat[0, :]
+
+        # No wakes: withwakes == background
+        self.wind_speeds_withwakes = self.wind_speeds_background.copy()
+
+        # For now require "wd_mean" to be in the df_scada
+        if "wd_mean" in df_scada.columns:
+            self.wd_mat_mean = df_scada["wd_mean"].values.astype(hercules_float_type)
+        else:
+            # Place holder for wind direction mean
+            self.wd_mat_mean = np.zeros(len(df_scada), dtype=hercules_float_type)
+
+        if "ti_000" in df_scada.columns:
+            self.ti_mat = df_scada[
+                [f"ti_{t_idx:03d}" for t_idx in range(self.n_turbines)]
+            ].to_numpy(dtype=hercules_float_type)
+
+            # Compute the turbine averaged turbulence intensities (axis = 1) using mean
+            self.ti_mat_mean = np.mean(self.ti_mat, axis=1, dtype=hercules_float_type)
+
+            self.initial_tis = self.ti_mat[0, :]
+
+        else:
+            self.ti_mat_mean = 0.08 * np.ones_like(self.ws_mat_mean, dtype=hercules_float_type)
+
+        # No wake deficits
+        self.floris_wake_deficits = np.zeros(self.n_turbines, dtype=hercules_float_type)
+
+        # Infer the rated power as the 99 percentile of the power column of 0th turbine
+        self.rated_turbine_power = np.percentile(df_scada[self.power_columns[0]], 99)
+
+        # Get the capacity of the farm
+        self.capacity = self.n_turbines * self.rated_turbine_power
+
+        self.logger.info(f"Inferred rated turbine power: {self.rated_turbine_power}")
+        self.logger.info(f"Inferred capacity: {self.capacity / 1e3} MW")
+
+        # Initialize the turbine array
+        self.turbine_array = TurbineFilterModelVectorizedSCADA(self.dt, self.scada_powers[0, :])
```

Comment:
Is `turbine_model_type` (or `turbine_file_name`) ignored for this type of wind model? Do we need to provide a warning or error if `turbine_model_type` is incorrectly specified?

Reply:
I have now added a warning that the turbine model here does not apply filtering. Plan to update at a later date.